app/configurator/interactive-filters/time-slider.tsx

```diff
@@ -0,0 +1,61 @@
+import { scaleLinear } from "d3";
+import React, { ChangeEvent } from "react";
+
+import { ChartState, useChartState } from "@/charts/shared/use-chart-state";
+import { useInteractiveFilters } from "@/charts/shared/use-interactive-filters";
+import { TableChartState } from "@/charts/table/table-state";
+import { Slider } from "@/components/form";
+import { parseDate } from "@/configurator/components/ui-helpers";
+
+export const TimeSlider = ({ componentIri }: { componentIri?: string }) => {
+  const [_, dispatch] = useInteractiveFilters();
+  const [t, setT] = React.useState(0);
+  const chartState = useChartState() as NonNullable<
+    Exclude<ChartState, TableChartState>
+  >;
+  const onChange = React.useMemo(() => {
+    // FIXME: enable interactive filters for maps!
+    if (componentIri && chartState.chartType !== "map") {
+      const rawRange = [
+        ...new Set(
+          chartState.allData.map((d) => d[componentIri]).filter(Boolean)
+        ),
+      ].sort() as string[];
+      const range = rawRange.map(parseDate);
+      const [min, max] = [range[0], range[range.length - 1]];
+      const scale = scaleLinear().range([min.getTime(), max.getTime()]);
+      const reversedRange = range.reverse();
+
+      return (e: ChangeEvent<HTMLInputElement>) => {
+        const t = +e.target.value;
+        setT(t);
+
+        const tDate = new Date(scale(t));
+        const IFDate = reversedRange.find((d) => d <= tDate) as Date;
```

Comment:
Since we have a sorted list here, I think we could have d3 bisect here to search for the nearest date. I think it is important to be fast here as we could have a lot of dates and the user could slide quickly.

Reply:
Yup, also for the time slider it would be good to only trigger dispatch when new date is different than current date, as right now we update unnecessarily – I'll modify this part, thanks for suggestions 🙇